ostd/src/mm/page_table/node.rs

```diff
@@ -231,8 +231,18 @@ where
     /// set the lock bit for performance as it is exclusive and unlocking is an
     /// extra unnecessary expensive operation.
     pub(super) fn alloc(level: PagingLevel) -> Self {
-        let meta = PageTablePageMeta::new_locked(level);
-        let page = page::allocator::alloc_single::<PageTablePageMeta<E, C>>(meta).unwrap();
+        let page = page::allocator::PAGE_ALLOCATOR.get().unwrap().lock().alloc_page(PAGE_SIZE).map(|paddr|{
+            Page::<PageTablePageMeta<E, C>>::from_unused(paddr)
```

Comment:
Firstly, the `page::allocator::alloc_single::<PageTablePageMeta<E, C>>(meta)` should be kept for allocating typed pages easier. Secondly, where's the `from_unused` function that receives initial metadata?